api/v1alpha1/externalip_types.go

```diff
@@ -42,11 +42,23 @@ type ExternalIPState string
 
 // All defined ExternalIPStates
 const (
-	ExternalIPStateNone       ExternalIPState = ""
-	ExternalIPStateReserved   ExternalIPState = "Reserved"
```

Comment:
I think the "Reserved" case should be kept, it is useful to denote ExternalIP that are available to be reused, and ones that are pending association.